packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -42,17 +129,8 @@ export function layerReducer(
 
       // We delete in reverse so that the indexPaths remain accurate even
       // after some layers are deleted.
-      const reversed = [...indexPaths.reverse()];
-
       return produce(state, (draft) => {
-        reversed.forEach((indexPath) => {
-          const childIndex = indexPath[indexPath.length - 1];
-          const parent = Layers.access(
-            draft.sketch.pages[pageIndex],
-            indexPath.slice(0, -1),
-          ) as Layers.ParentLayer;
-          parent.layers.splice(childIndex, 1);
-        });
+        deleteLayers(indexPaths, draft.sketch.pages[pageIndex]);
```

Comment:
There's a comment above this that should go into `deleteLayer`

```ts
// We delete in reverse so that the indexPaths remain accurate even
// after some layers are deleted.
```